### PR TITLE
Fix: a new formatter library (issue #2601 and #2606)

### PR DESCRIPTION
**Pull request by kirk0830**
To systematically solve formatting related issue, a new library now is available. To use it, please refer to online FeiShu document (open access): [Instruction of ABACUS formatter library](https://ucoyxk075n.feishu.cn/docx/Yym9dnm3aoTMfHxin8rcX9Rvnmb).

### CODE REVIEW

**Comment by kirk0830** [comment]
issue #2601 #2606

File: source/module_base/test/formatter_contextfmt_test.cpp [hunk 1]
```diff
@@ -0,0 +1,369 @@
+#include "../formatter.h"
+#include <gtest/gtest.h>
+
```
Comment (hongriTianqi) [inline source/module_base/test/formatter_contextfmt_test.cpp:3]: Please add some comments of the unit tests. Thanks!

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: source/module_base/test/formatter_contextfmt_test.cpp (hunk 1 above)
Comment (kirk0830) [reply source/module_base/test/formatter_contextfmt_test.cpp:3]: Not at all, I have completed adding short instructions in each unittest file

**Comment by kirk0830** [comment]
later when I plan to make changes I will create new branch to get rid of previous submitted PR info.

**Review by hongriTianqi — APPROVED** [review]

**Comment by hongriTianqi** [comment]
> later when I plan to make changes I will create new branch to get rid of previous submitted PR info.

This is not needed, because the merge option can delete all the commit info.

**Pull request merged**